### ISSUE TITLE
Add lint of a null check then spread or expand `prefer_spread_collections`

### ISSUE THREAD

**Issue by parlough** [dart-lang/linter#5093]
## Name

Some options:

- `verbose_conditional_collection_spread`
- `unnecessary_null_check_and_spread`
  - Matches the naming scheme of other lints and diagnostics
- `prefer_null_aware_spread_operator`

## Description

The analyzer produces this diagnostic when a list is added to a collection literal with a spread operator after checking if it's not null.

## Categories

`style`, `languageFeatureUsage`

## Bad Examples

```dart
final list = [
  itemOne,
  itemTwo,
  if (extraItems != null) ...extraItems,
];
```

## Good Examples 

```dart
final list = [
  itemOne,
  itemTwo,
  ...?extraItems,
];
```

## Fix discussion

A widely applicable fix should be possible. If the lint isn't seen as valuable enough to many, just the assist would still be valuable.

## Alternative

Expand [`prefer_spread_collections`](https://dart.dev/lints/prefer_spread_collections) to cover these cases as it already covers some null-aware cases, and I imagine if one wants that, they'd almost always prefer to use it here too.

**Comment by srawlins** [comment]
I'd be in favor of enhancing prefer_spread_collections.
